fireflower/core.py

```diff
@@ -79,11 +84,13 @@ def wrapper(self, *args, **kwargs):
                     'task_kwargs': self.param_kwargs,
                 }
                 FireflowerStateManager.sentry.captureException(extra=extra)
+                FireflowerStateManager.dogstatsd.close_buffer()
             raise
         finally:
             if (FireflowerStateManager.sentry and
                     FireflowerStateManager.sentry.client):
                 FireflowerStateManager.sentry.client.context.clear()
+                FireflowerStateManager.dogstatsd.close_buffer()
```

Comment:
Condition on `if FireflowerStateManager.dogstatsd is not None`, since callers might not have a statsd to register and that's ok.